scipy/sparse/linalg/_eigen/tests/test_svds.py

```diff
@@ -721,6 +717,9 @@ def test_svd_LM_ones_matrix(self, shape, dtype):
             U, s, VH = svds(A, k, solver=self.solver)
 
         # Check some generic properties of svd.
+        if (self.solver == 'arpack' and dtype is complex):
+            pytest.skip("ARPACK has additional restriction for complex dtype")
```

Comment:
We can't do this. If it doesn't work then we have to fix this issue first. We don't have the luxury to push the problems to the users. It doesn't matter if it is related to this PR or not. If there is an issue with ARPACK then this PR is blocked by that problem and until it is resolved we can't proceed with others. There are many PRs waiting in that state.

Reply:
@ilayn - thank you for your prompt review!
These two lines just basically copy/paste the already skipped complex ARPACK tests in lines 568-569 as well as in lines 743-745. So this is not a new issue - complex ARPACK tests already failed before, - but it has not blocked merging unrelated PRs in the past. 
There are already apparently relevant issues open, e.g., https://github.com/scipy/scipy/issues/13692, https://github.com/scipy/scipy/issues/3716
that hopefully would be addressed at some point, but they do not prevent from making unrelated changes in svds,

Reply:
Lately, we have been ignoring almost every issue regarding the stability of these in the name of "let's do it in a later PR" and apologies for being so blunt but this module is going down in terms of quality with that kind of behavior. It's collecting lots and lots of edge cases where things are broken and waiting to be picked up again that never happens. But following them up are very difficult because it is full of ignored tests and other comments that are dispersed in many PRs. 

If ARPACK is not working properly or showing its age, then deprecate it. If it is working but we can't make it work then let's fix that part but leaving some solvers in broken state is not going to help anyone. 

I am having difficulty already finding time for contributing to SciPy and doing passing-by reviews and I really don't want to continuously argue with sparse contributors about this. I'll leave it to other maintainers to decide.

Reply:
I second @ilayn's comment. If we have better/more robust solvers, then let's remove this one.

Reply:
This is not an ARPACK issue. svds(solver="arpack") calls eigsh to get eigenvectors of A @ A.T.conj(). However ARPACK has no restarted Lanczos algorithm for complex hermitian matrices, therefore in the case of complex matrices, eigsh calls something else. 

At any rate, it's a known issue (judging by already skipped unit tests) and clearly unrelated to this PR.

Reply:
@tupui @ilayn This discussion is unrelated to the given PR. I propose that the discussion is moved to https://github.com/scipy/scipy/issues/13692#issuecomment-1159782445 where the issue has been reported. I've just made a corresponding comment there.

I also propose that this PR is merged, so that we could move on to fixing the next issue with the script, e.g., https://github.com/scipy/scipy/issues/16330